go/internal/cbor/deterministic_test.go

```diff
@@ -0,0 +1,174 @@
+package cbor
+
+import (
+	"testing"
+)
+
+var (
+	// Deterministically encoded unsigned integers.
+	uint0          = []byte{0x00}
+	uint10         = []byte{0x0A}
+	uint23         = []byte{0x17}
+	uint24         = []byte{0x18, 0x18}
+	uint45         = []byte{0x18, 0x2D}
+	uint255        = []byte{0x18, 0xFF}
+	uint256        = []byte{0x19, 0x01, 0x00}
+	uint5000       = []byte{0x19, 0x13, 0x88}
+	uint65535      = []byte{0x19, 0xFF, 0xFF}
+	uint65536      = []byte{0x1A, 0x00, 0x01, 0x00, 0x00}
+	uint4294967    = []byte{0x1A, 0x00, 0x41, 0x89, 0x37}
+	uint4294967295 = []byte{0x1A, 0xFF, 0xFF, 0xFF, 0xFF}
+	uint4294967296 = []byte{0x1B, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00}
+
+	// Long.MAX_VALUE aka 0x7F 0xFF 0xFF 0xFF 0xFF 0xFF 0xFF 0xFF.
+	uint9223372036854775807 = []byte{0x1B, 0x7F, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF}
+
+	// Max CBOR supported value 0xFF 0xFF 0xFF 0xFF 0xFF 0xFF 0xFF 0xFF.
+	uint18446744073709551615 = []byte{0x1B, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF}
+
+	uint64TestCasesAsBytes = [][]byte{uint0, uint10, uint23, uint24, uint45, uint255, uint256, uint5000, uint65535, uint65536, uint4294967, uint4294967295, uint4294967296, uint9223372036854775807, uint18446744073709551615}
+	uint64Values           = []uint64{0, 10, 23, 24, 45, 255, 256, 5000, 65535, 65536, 4294967, 4294967295, 4294967296, 9223372036854775807, 18446744073709551615}
+)
+
+func TestUintDeterministic(t *testing.T) {
+	// Single deterministically encoded unsigned integers return true.
+	for _, testCase := range uint64TestCasesAsBytes {
+		err := Deterministic(testCase)
+		if err != nil {
+			t.Error("Deterministically encoded unsigned integers should not return error.")
+		}
+	}
+
+	// Also deterministically encoded unsigned integers following each other return true.
+	concatenatedTestCases := multiappend(uint64TestCasesAsBytes)
+	err := Deterministic(concatenatedTestCases)
+	if err != nil {
+		t.Error("Deterministically encoded unsigned integers should not return false for deterministicy.")
+	}
+}
+
```

Comment:
Can you add a test case where the input to `Deterministic` is a concatenation of two unsigned integers?

Reply:
There is a test where there is a concatenation of all the unsigned integers used in the example. I think this should cover that even better? See TestUintDeterministic second part. I could separate this to it's own test though if that would be clearer?